java/src/main/java/com/nedap/retail/example/rest/ApiCaller.java

```diff
@@ -32,45 +35,53 @@
 
     public ApiCaller(final String baseUrl) {
         // make sure baseUrl does not end with a slash
-        if (baseUrl.endsWith("/")) {
-            this.baseUrl = baseUrl.substring(0, baseUrl.length() - 1);
-        } else {
-            this.baseUrl = baseUrl;
+        this.baseUrl = baseUrl;
```

Comment:
If the `ApiCaller` does not work with a trailing slash, maybe it's a good idea to throw a (runtime) exception (e.g. `IllegalArgumentException`) to indicate a programming error. It is very likely an integrator will just copy-paste the `ApiCaller` class.